elrond-config.md

```diff
@@ -715,110 +617,95 @@ TODO: Implement [reserved keys and read-only runtimes](https://github.com/Elrond
          </commands>
       [priority(60)]
 
-    // The memory instance linked to the contract's account should be restored to the initial state after the contract call.
-    // TODO Create a new wasm instance to run the contract call. Execute all contract calls on a separate wasm instance.
-    //      This will eliminate the need for #restoreMem
-    //          https://github.com/multiversx/mx-chain-vm-go/blob/255d2b23189cbc8a80312ab89890163800255dba/vmhost/hostCore/execution.go#L779
-    //      Push the current <callState>, <bigIntHeap>, <bufferHeap> and wasm instance to a call stack (add new configuration cell <callStack>)       
-    rule <commands> callContract(FROM, TO, VALUE, ESDT, FUNCNAME:WasmStringToken, ARGS, GASLIMIT, GASPRICE)
-                 => pushWorldState
-                 ~> transferFunds(FROM, TO, VALUE)
-                 ~> transferESDTs(FROM, TO, ESDT)
-                 ~> mkCall(FROM, TO, VALUE, ESDT, FUNCNAME, ARGS, GASLIMIT, GASPRICE)
-                 ~> #endWasm
-                 ~> #restoreMem(MEMADDR, MMAX, SIZE, DATA)
-                    ...
-         </commands>
-         <account>
-           <address> TO </address>
-           <codeIdx> MODIDX:Int </codeIdx>
-           ...
-         </account>
-         <moduleInst>
-           <modIdx> MODIDX </modIdx>
-           <memAddrs> 0 |-> MEMADDR </memAddrs>
-           ...
-         </moduleInst>
-         <memInst>
-           <mAddr> MEMADDR </mAddr>
-           <mmax> MMAX </mmax>
-           <msize> SIZE </msize>
-           <mdata> DATA </mdata>
-         </memInst>
-      [priority(60)]
+    rule [callContract]:
+        <commands> callContract(FROM, TO, VALUE, ESDT, FUNCNAME:WasmStringToken, ARGS, GASLIMIT, GASPRICE)
+                => pushWorldState
+                ~> pushCallState
+                ~> transferFunds(FROM, TO, VALUE)
+                ~> transferESDTs(FROM, TO, ESDT)
+                ~> newWasmInstance(CODE)
+                ~> mkCall(FROM, TO, VALUE, ESDT, FUNCNAME, ARGS, GASLIMIT, GASPRICE)
+                ~> #endWasm
+                   ...
+        </commands>
+        <account>
+          <address> TO </address>
+          <code> CODE </code>
+          ...
+        </account>
+        // output
+        <out> _ => .List </out>
```

Comment:
I'm trying to understand this rule, and I noticed that it is only used in mandos tests. Is it also intended to be used for normal calls between contracts?

If the latter is true, should all of these cells be cleared? I'm not saying that it's wrong, but I would like to make sure that this is the actually desired semantics, i.e. a contract can't build some of these cells incrementally (e.g. the message), and build part of it before calling another contract, and part of it after.